src/parser.cpp

```diff
@@ -455,28 +443,27 @@ namespace ada::parser {
           size_t consumed_bytes = url.parse_port(port_view, true);
           input_position += consumed_bytes;
           if(!url.is_valid) { return url; }
-          goto goto_path_start;
+          state = state::PATH_START;
+          [[fallthrough]];
         }
         case ada::state::PATH_START: {
-          goto_path_start:
           ada_log("PATH_START ", helpers::substring(url_data, input_position));
+
           // If url is special, then:
           if (url.is_special()) {
-
             // Set state to path state.
             state = ada::state::PATH;
 
             // Optimization: Avoiding going into PATH state improves the performance of urls ending with /.
-            if (input_position == input_size) {
+            if (input_position == input_size && url_data[input_position] == '/') {
```

Comment:
This is certainly wrong. We are saying that if input_position == input_size, then we want to load `url_data[input_size]` which is unsafe.

Reply:
Here you are changing the logic of the code in a way that I am not sure I understand.

Reply:
Yes you're right. I'm updating it.